---
created_at: 2026-01-15
description: Information about databases on Mahuika
tags:
    - database
    - biology
---

Many research projects use reference or external databases.
This page describes databases that exist on Mahuika for use as well as recommendations for using some specific external databases.

## Maintained databases on Mahuika

Some databases are readable for all users on Mahuika.
These databases can be found at `/opt/nesi/db`.
Some environmental modules depend on these databases and connect to these directories automatically.

Dataset | Path | Licence Status | Notes
-- | -- | -- | --
[alphafold_db](https://alphafold.ebi.ac.uk/) | /opt/nesi/db/alphafold_db | CC-BY-4.0 | Predicted protein structures generated by AlphaFold
[blast](https://blast.ncbi.nlm.nih.gov/) | /opt/nesi/db/blast | Public | NCBI BLAST nucleotide and protein databases
[cartopy](https://github.com/SciTools/cartopy) | /opt/nesi/db/cartopy | BSD-3-Clause | Databases for cartopy module
[centrifuge](https://ccb.jhu.edu/software/centrifuge/) | /opt/nesi/db/centrifuge | GPL-3.0 | Databases for centrifuge module
[CheckM2_DB](https://github.com/chklovski/CheckM2) | /opt/nesi/db/CheckM2_DB | GPL-3.0 | Database for CheckM2 module
[CheckM_DB](https://github.com/Ecogenomics/CheckM) | /opt/nesi/db/CheckM_DB | GPL-3.0 | Database for CheckM module
[DAS_DB](https://github.com/cmks/DAS_Tool) | /opt/nesi/db/DAS_DB | [Public](https://github.com/cmks/DAS_Tool/blob/master/license.txt) | Database for DAS_Tool module
[dammit_db](https://github.com/dib-lab/dammit) | /opt/nesi/db/dammit_db | BSD | Databases for dammit module
[dfam_3.9](https://dfam.org/) | /opt/nesi/db/dfam_3.9 | CC0 1.0 | Transposable element profile HMM database
[DRAM_1.3.5](https://github.com/WrightonLabCSU/DRAM) | /opt/nesi/db/DRAM_1.3.5 | GPL-3.0 | Databases for DRAM module
[eggnog_db](https://eggnogdb.org/) | /opt/nesi/db/eggnog_db | Unspecified | Orthologous group and functional annotation database
[FCS-GX](https://github.com/ncbi/fcs) | /opt/nesi/db/FCS-GX | United States Government Work | Database for FCS-GX module
[gtdbtk_202](https://gtdb.ecogenomic.org/) | /opt/nesi/db/gtdbtk_202 | CC-BY-SA 4.0 | Genome Taxonomy Database release 202 used by GTDB-Tk module
[gtdbtk_207_v2](https://gtdb.ecogenomic.org/) | /opt/nesi/db/gtdbtk_207_v2 | CC-BY-SA 4.0 | Genome Taxonomy Database release 207 v2 used by GTDB-Tk module
[gtdbtk_214](https://gtdb.ecogenomic.org/) | /opt/nesi/db/gtdbtk_214 | CC-BY-SA 4.0 | Genome Taxonomy Database release 214 used by GTDB-Tk module
[gtdbtk_220](https://gtdb.ecogenomic.org/) | /opt/nesi/db/gtdbtk_220 | CC-BY-SA 4.0 | Genome Taxonomy Database release 220 used by GTDB-Tk module
[HUMAnN](https://huttenhower.sph.harvard.edu/humann/) | /opt/nesi/db/Humann | MIT | Databases for HUMAnN module
[Kaiju](https://github.com/bioinformatics-centre/kaiju) | /opt/nesi/db/Kaiju | GPL-3.0 | Database index for Kaiju module
[Kraken2](https://github.com/DerrickWood/kraken2) | /opt/nesi/db/Kraken2 | MIT | Databases for Kraken2 module
[megaX](https://www.megasoftware.net/) | /opt/nesi/db/megaX | Free for academics | Evolutionary analysis reference data for MegaX module
[nullarbor_db](https://github.com/tseemann/nullarbor) | /opt/nesi/db/nullarbor_db | GPL-2.0 | Reference databases used by the Nullarbor module
[Pfam](https://www.ebi.ac.uk/interpro/download/pfam/) | /opt/nesi/db/Pfam | CC0 | Protein family HMM database
[PhyloPhlAn](https://github.com/biobakery/phylophlan) | /opt/nesi/db/PhyloPhlAn | MIT | Databases of universal markers for prokaryotes
[prokka](https://github.com/tseemann/prokka) | /opt/nesi/db/prokka | GPL-3.0 | Databases needed for prokka environmental module
[ProteinDataBank](https://www.rcsb.org/) | /opt/nesi/db/ProteinDataBank | CC0 1.0 | 3D structures of proteins and nucleic acids
[RQCFilterData](https://github.com/microbiomedata/ReadsQC) | /opt/nesi/db/RQCFilterData | Unspecified | Reference data for read quality control
[sortmerna_db](https://github.com/sortmerna/sortmerna) | /opt/nesi/db/sortmerna_db | GPL-3.0 | rRNA reference databases for SortMeRNA module
[SqueezeMeta](https://github.com/jtamames/SqueezeMeta) | /opt/nesi/db/SqueezeMeta | GPL-3.0 | Databases needed for SqueezeMeta module
[StrVCTVRE](https://github.com/andrewSharo/StrVCTVRE) | /opt/nesi/db/StrVCTVRE | MIT | Training and reference data for StrVCTVRE module
[Trinotate](https://github.com/Trinotate/Trinotate) | /opt/nesi/db/Trinotate | Public | Databases needed for Trinotate module
[Uniprot](https://www.uniprot.org/) | /opt/nesi/db/Uniprot | CC-BY 4.0 | Protein sequence and functional annotation database
[VEP](https://www.ensembl.org/info/docs/tools/vep/) | /opt/nesi/db/VariantEffectPredictor | No restrictions | Ensembl annotation data for variant effect prediction
[VIBRANT_v1.2.1_databases](https://github.com/AnantharamanLab/VIBRANT) | /opt/nesi/db/VIBRANT_v1.2.1_databases | GPL-3.0 | Viral genome and HMM databases used by VIBRANT environmental module
[VirSorter](https://github.com/simroux/VirSorter) | /opt/nesi/db/VirSorter | GPL-2.0 | Viral hallmark gene and profile databases
[waafle](https://github.com/biobakery/waafle) | /opt/nesi/db/waafle | MIT | Reference sets for gene neighborhood analysis for waafle environmental module
[checkv-db-v0.6](https://bitbucket.org/berkeleylab/checkv) | /opt/nesi/db/checkv-db-v0.6 | [BSD 3-Clause-style](https://bitbucket.org/berkeleylab/checkv/src/master/LICENSE.txt) | Viral genome completeness and contamination database for CheckV environmental module

There are also some versioned databases which are accessible through environmental modules, specifically:

- AlphaFold2DB: `module avail AlphaFold2DB`
- AlphaFold3DB: `module avail AlphaFold3DB`
- BLASTDB: `module avail BLASTDB`

!!! note "Requesting new or updated databases"
    If there is a database you think may be useful to many Mahuika users, or if you would like an updated version of one of the maintained databases, please {% include "partials/support_request.html" %} with details about the source and version of the database of interest.

## Recommendations for obtaining data from selected external databases

### JGI Portals

The [Joint Genome Institute](https://jgi.doe.gov/) has many databases and data portals available.
To download/access files from JGI you will need to register for an account.
We recommend you utilize the [Globus endpoint provided by JGI](https://genome.jgi.doe.gov/portal/help/download.jsf#/globus) to directly transfer files from the JGI servers to Mahuika.
For more information about using Globus on Mahuika see [the Globus docs section](../Data_Transfer/Globus/Overview.md).

### NCBI

We have severak environmental modules to aid in finding and downloading data from NCBI:

- the [NCBI Datasets command-line tools](https://www.ncbi.nlm.nih.gov/datasets/docs/v2/command-line-tools/download-and-install/): `module avail datasets`
- the [EDirect command line interface](https://www.ncbi.nlm.nih.gov/books/NBK179288/) with the [Entrez search system](https://www.ncbi.nlm.nih.gov/books/NBK3837/): `module avail entrez-direct`
- the [SRA Toolkit](https://github.com/ncbi/sra-tools): `module avail sratoolkit`
